tests/python/rest_api/test_tasks.py

```diff
@@ -762,6 +763,53 @@ def test_can_create_task_with_exif_rotated_images(self):
                         # original is 480x640 with 90/-90 degrees rotation
                         assert im.height == 640 and im.width == 480
 
+    def test_can_create_task_with_big_images(self):
+        # Checks for regressions about the issue
+        # https://github.com/opencv/cvat/issues/6878
+        # In the case of big files (>2.5 MB by default),
+        # uploaded files could be write-appended twice,
+        # leading to bigger raw file sizes than expected.
+
+        task_spec = {
+            "name": f"test {self._USERNAME} to create a task with big images",
+            "labels": [
+                {
+                    "name": "car",
+                }
+            ],
+        }
+
+        # We need a big file to reproduce the problem
+        image_file = generate_image_file("big_image.bmp", size=(4000, 4000), color=(100, 200, 30))
+        file_size = len(image_file.getvalue())
+        assert 10 * 2**20 < file_size
+
+        task_data = {
+            "client_files": [image_file],
+            "image_quality": 70,
+            "use_cache": True,
+            "use_zip_chunks": True,
+        }
+
+        task_id, _ = create_task(self._USERNAME, task_spec, task_data)
+
+        # check that the uploaded unprocessed files have the original size
+        script = textwrap.dedent(
+            """\
+            import os
+            from cvat.apps.engine.models import Data
+            data = Data.objects.get(tasks__id={task_id})
+            print(os.stat(f"data/data/{{data.id}}/raw/{image_file_name}").st_size)
```

Comment:
Wouldn't it be possible to check this by requesting the original chunk via the API and examining it? Seems like it would be more robust than peeking inside the container.

Reply:
It can work, I need to check.

Reply:
Updated